File: src/main/java/com/rover12421/csdn/java2kt/java/_5_MultiLoopJump.java
package com.rover12421.csdn.java2kt.java;

import java.io.*;

/**
 * Created by rover12421 on 8/10/16.
 */
public class _5_MultiLoopJump {
    public static void main(String[] args) {
        int max = 10;
        while (max-- > 0) {
            try(
                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
                    PrintStream ps = new PrintStream(baos)
                    ) {
                try {
                    if (max == 2) {
                        break;
                    }
                }catch (Throwable e2) {
                    e2.printStackTrace();
                }
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        System.out.println("max == " + max);
    }
}

File: src/main/kotlin/com/rover12421/csdn/java2kt/kt/_5_MultiLoopJump.kt
package com.rover12421.csdn.java2kt.kt

import java.io.*

/**
 * Created by rover12421 on 8/10/16.
 */
object _5_MultiLoopJump {
    @JvmStatic fun main(args: Array<String>) {
        var max = 10
        var finish = false
        while (!finish && max-- > 0) while1@ {
            try {
                 ByteArrayOutputStream().use use1@ { baos ->
                    PrintStream(baos).use use2@ { ps ->
                        try {
                            if (max == 2) {
                                //没法用break跳出while循环
//                                break
                                //return 也没法直接return到while
//                                return
                                finish = true
                            }
                        } catch (e2: Throwable) {
                            e2.printStackTrace()
                        }
                    }
                }
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }

        println("max == " + max)
    }
}
